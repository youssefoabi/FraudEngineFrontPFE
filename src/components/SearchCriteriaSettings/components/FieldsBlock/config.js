export const ActivateFieldsConfig = [
  {
    columns: 6,
    id: 'isEnable',
    type: 'switch',
    label: 'Actif',
    defaultValue: false,
  },
  {
    columns: 6,
    id: 'isMandatory',
    type: 'switch',
    label: 'Champ obligatoire',
    defaultValue: false,
  },
];

export const PlaceholderConfig = [
  {
    columns: 6,
    id: 'label',
    type: 'input',
    label: 'Placeholder',
  },
];

export const typesOptions = [
  { value: 'Address', fieldTypeId: 1, label: "Autocomplétion d'adresse" },
  { value: 'Numeric', fieldTypeId: 3, label: 'Numérique' },
  { value: 'Alphanumeric', fieldTypeId: 2, label: 'Texte libre' },
];

export const FIELDS_BY_ID = { Address: 1, Numeric: 3, Alphanumeric: 2 };
