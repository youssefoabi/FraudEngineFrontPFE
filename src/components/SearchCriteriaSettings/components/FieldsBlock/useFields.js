import { pathOr, prop } from 'ramda';

const useFields = ({ handleChange, engine, fieldId }) => {
  const { currentScope } = engine || {};
  const inputFields = pathOr({}, ['inputFields'], engine);
  const currentInputField = pathOr({}, [fieldId], inputFields);

  const handleChangeType = ({ value }) => {
    const type = pathOr(null, ['value'], value);
    const currentType = pathOr({}, [fieldId, 'type', 'value'], inputFields);

    if (type !== currentType) {
      const parameters = pathOr([], ['parameters'], currentInputField);
      const newParameters = parameters.filter(
        param => prop('scopeId', param) !== prop('scopeId', currentScope),
      );
      handleChange({
        id: 'inputFields',
        value: {
          ...inputFields,
          [fieldId]: {
            ...currentInputField,
            type: value,
            parameters: newParameters,
          },
        },
      });
    }
  };

  const handleFieldsChange = ({ id, value }) => {
    handleChange({
      id: 'inputFields',
      value: { ...inputFields, [fieldId]: { ...currentInputField, [id]: value } },
    });
  };

  return { handleFieldsChange, inputFields, currentScope, handleChangeType };
};

export default useFields;
