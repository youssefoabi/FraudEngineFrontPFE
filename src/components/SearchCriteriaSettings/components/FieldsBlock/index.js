import { DataEdit, GridItem } from '@cdiscount/backoffice-ui';
import { func, number, shape } from 'prop-types';
import { Grid } from '@material-ui/core';
import { isNil, pathOr } from 'ramda';
import React from 'react';

import { ActivateFieldsConfig, PlaceholderConfig, typesOptions } from './config';
import useFields from './useFields';

const FieldsBlock = ({ engine, handleChange, fieldId }) => {
  const { handleFieldsChange, handleChangeType, inputFields } = useFields({
    engine,
    handleChange,
    fieldId,
  });

  const currentInputField = pathOr(null, [fieldId], inputFields);
  if (isNil(fieldId)) return null;

  return (
    <GridItem title={`Champ de saisie ${fieldId + 1}`}>
      <Grid item xs={12} md={12} lg={12}>
        <DataEdit fields={ActivateFieldsConfig} values={currentInputField} onChange={handleFieldsChange} />
        <DataEdit
          fields={[
            {
              columns: 6,
              id: `type`,
              type: 'select',
              label: 'Type',
              selectProps: {
                options: typesOptions,
              },
            },
          ]}
          values={currentInputField}
          onChange={handleChangeType}
        />
        <br />
        <DataEdit fields={PlaceholderConfig} values={currentInputField} onChange={handleFieldsChange} />
      </Grid>
    </GridItem>
  );
};

FieldsBlock.propTypes = {
  engine: shape({}),
  handleChange: func,
  fieldId: number,
};

FieldsBlock.defaultProps = {
  engine: null,
  handleChange: Function.Prototype,
  fieldId: null,
};

export default FieldsBlock;
