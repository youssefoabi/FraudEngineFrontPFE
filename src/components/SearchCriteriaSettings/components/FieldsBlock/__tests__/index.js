import { itRendersAllMutations } from '@cdiscount/jest-utils';

import FieldsBlock from '../';

const mutations = [
  { name: 'without props', props: {} },
  { name: 'with fieldId null', props: { fieldId: null } },
  { name: 'with fieldId equal 1', props: { fieldId: 1 } },
];

describe(FieldsBlock.name, () => {
  itRendersAllMutations(FieldsBlock, mutations);
});
