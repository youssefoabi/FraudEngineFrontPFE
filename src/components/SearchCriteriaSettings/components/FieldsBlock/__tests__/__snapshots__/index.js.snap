// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldsBlock FieldsBlock renders correctly with fieldId equal 1 1`] = `
<h
  fullheight={false}
  title="Champ de saisie 2"
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    lg={12}
    md={12}
    xs={12}
  >
    <f
      config={null}
      fields={
        Array [
          Object {
            "columns": 6,
            "defaultValue": false,
            "id": "isEnable",
            "label": "Actif",
            "type": "switch",
          },
          Object {
            "columns": 6,
            "defaultValue": false,
            "id": "isMandatory",
            "label": "Champ obligatoire",
            "type": "switch",
          },
        ]
      }
      inputs={null}
      onChange={[Function]}
      onInit={[Function]}
      title={null}
      values={null}
    />
    <f
      config={null}
      fields={
        Array [
          Object {
            "columns": 6,
            "id": "type",
            "label": "Type",
            "selectProps": Object {
              "options": Array [
                Object {
                  "fieldTypeId": 1,
                  "label": "Autocomplétion d'adresse",
                  "value": "Address",
                },
                Object {
                  "fieldTypeId": 3,
                  "label": "Numérique",
                  "value": "Numeric",
                },
                Object {
                  "fieldTypeId": 2,
                  "label": "Texte libre",
                  "value": "Alphanumeric",
                },
              ],
            },
            "type": "select",
          },
        ]
      }
      inputs={null}
      onChange={[Function]}
      onInit={[Function]}
      title={null}
      values={null}
    />
    <br />
    <f
      config={null}
      fields={
        Array [
          Object {
            "columns": 6,
            "id": "label",
            "label": "Placeholder",
            "type": "input",
          },
        ]
      }
      inputs={null}
      onChange={[Function]}
      onInit={[Function]}
      title={null}
      values={null}
    />
  </WithStyles(ForwardRef(Grid))>
</h>
`;

exports[`FieldsBlock FieldsBlock renders correctly with fieldId null 1`] = `""`;

exports[`FieldsBlock FieldsBlock renders correctly without props 1`] = `""`;
